File: content/api-documentation/how-to/market-data/examples/getPrices.js
const moment = require("moment");
const Alpaca = require("@alpacahq/alpaca-trade-api");
const alpaca = new Alpaca();

//Get daily price data for AAPL over the last 7 days.
let bars = alpaca.getBarsV2(
  "AAPL",
  {
    start: moment().subtract(7, "days").format(),
    end: moment().subtract(20, "minutes").format(),
    timeframe: "1Day",
  },
  alpaca.configuration
);
const barset = [];

for await (let b of bars) {
  barset.push(b);
}

const week_open = barset[0].OpenPrice;
const week_close = barset.slice(-1)[0].ClosePrice;
const percent_change = ((week_close - week_open) / week_open) * 100;

console.log(`AAPL moved ${percent_change}% over the last 7 days`);

File: content/api-documentation/how-to/market-data/examples/getPrices.go
package main

import (
	"fmt"
	"time"

	"github.com/alpacahq/alpaca-trade-api-go/alpaca"
	"github.com/alpacahq/alpaca-trade-api-go/v2"
)

func init() {
	alpaca.SetBaseUrl("https://data.alpaca.markets")
}

func main() {
	bars := alpaca.GetBars(
		"AAPL", v2.Day, v2.Raw, time.Now().Add(-7*24*time.Hour), time.Now().Add(-20*time.Minute), 7)
	var barset []v2.Bar

	for bar := range bars {
		if bar.Error != nil {
			panic(bar.Error)
		}
		barset = append(barset, bar.Bar)
	}

	// See how much AAPL moved in that timeframe.
	startPrice := barset[0].Open
	endPrice := barset[len(barset)-1].Close
	percentChange := ((endPrice - startPrice) / startPrice ) * 100
	fmt.Printf("AAPL moved %v%% over the last 7 days.", percentChange)
}